docs/identifiers.md

```diff
@@ -1,8 +1,10 @@
 # Identifiers
-All objects in the Kubernetes REST API are identified by a Name and a UID.
+All objects in the Kubernetes REST API are unambiguously identified by a Name and a UID.
+
+For non-unique user-provided attributes, Kubernetes provides [labels](labels.md) and [annotations](annotations.md).
 
 ## Names
-Names are user-provided.  Only one object of a given kind can have a given name at a time.  But if you delete an object, you can make a new object with the same name.  Names are the used to refer to an object in a resource URL, such as `/api/v1beta3/pods/some.name`.   Names may be up to maximum length of 253 characters and consist of lower case alphanumeric characters, `-`, and `.`.  See the [identifiers design doc](design/identifiers.md) for the precise syntax rules for names.
+Names are generally client-provided.  Only one object of a given kind can have a given name at a time (i.e., they are spatially unique).  But if you delete an object, you can make a new object with the same name.  Names are the used to refer to an object in a resource URL, such as `/api/v1beta3/pods/some-name`.   Names may be up to maximum length of 253 characters and consist of lower case alphanumeric characters, `-`, and `.`.  See the [identifiers design doc](design/identifiers.md) for the precise syntax rules for names.
```

Comment:
There are cases where certain resources (users, tokens, complex virtual types) have different name restrictions.  I would suggest we say "by convention the names of Kubernetes resources should be ..."